internal/component/loki/secretfilter/secretfilter.go

```diff
@@ -0,0 +1,351 @@
+	ExcludeGeneric bool                `alloy:"exclude_generic,attr,optional"` // Exclude the generic API key rule (default: false)
+	AllowList      []string            `alloy:"allowlist,attr,optional"`       // List of regexes to allowlist (on top of what's in the Gitleaks config)
+	PartialMask    uint                `alloy:"partial_mask,attr,optional"`    // Show the first N characters of the secret (default: 0)
+}
+
+// Exports holds the values exported by the loki.secretfilter component.
+type Exports struct {
+	Receiver loki.LogsReceiver `alloy:"receiver,attr"`
+}
+
+// DefaultArguments defines the default settings for log scraping.
+var DefaultArguments = Arguments{}
+
+// SetToDefault implements syntax.Defaulter.
+func (args *Arguments) SetToDefault() {
+	*args = DefaultArguments
+}
+
+var (
+	_ component.Component     = (*Component)(nil)
+	_ component.LiveDebugging = (*Component)(nil)
+)
+
+// Component implements the loki.secretfilter component.
+type Component struct {
+	opts component.Options
+
+	mut       sync.RWMutex
+	args      Arguments
+	receiver  loki.LogsReceiver
+	fanout    []loki.LogsReceiver
+	Rules     []Rule
+	AllowList []AllowRule
+
+	debugDataPublisher livedebugging.DebugDataPublisher
+}
+
+// Non-exhaustive representation. See https://github.com/gitleaks/gitleaks/blob/master/config/config.go
+type GitLeaksConfig struct {
+	AllowList struct {
+		Description string
+		Paths       []string
+		Regexes     []string
+	}
+	Rules []struct {
+		ID          string
+		Description string
+		Regex       string
+		Keywords    []string
+		SecretGroup int
+
+		Allowlist struct {
+			StopWords []string
+			Regexes   []string
+		}
+	}
+}
+
+// New creates a new loki.secretfilter component.
+func New(o component.Options, args Arguments) (*Component, error) {
+	debugDataPublisher, err := o.GetServiceData(livedebugging.ServiceName)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &Component{
+		opts:               o,
+		receiver:           loki.NewLogsReceiver(),
+		debugDataPublisher: debugDataPublisher.(livedebugging.DebugDataPublisher),
+	}
+
+	// Call to Update() once at the start.
+	if err := c.Update(args); err != nil {
+		return nil, err
+	}
+
+	// Immediately export the receiver which remains the same for the component
+	// lifetime.
+	o.OnStateChange(Exports{Receiver: c.receiver})
+
+	return c, nil
+}
+
+// Run implements component.Component.
+func (c *Component) Run(ctx context.Context) error {
+	componentID := livedebugging.ComponentID(c.opts.ID)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case entry := <-c.receiver.Chan():
+			// Start processing the log entry to redact secrets
+			newEntry := c.processEntry(entry)
+			if c.debugDataPublisher.IsActive(componentID) {
+				c.debugDataPublisher.Publish(componentID, fmt.Sprintf("incoming entry: %s, redacted entry: %s", entry.Line, newEntry.Line))
```

Comment:
```suggestion
				c.debugDataPublisher.Publish(componentID, fmt.Sprintf("%s => %s", entry.Line, newEntry.Line))
```
nit: I would make it a bit less verbose and re-use the same arrow pattern as the loki relabel component